kivy/graphics/vertex_instructions.pyx

```diff
@@ -1123,7 +1123,8 @@ cdef class Ellipse(Rectangle):
     :Parameters:
         `segments`: int, defaults to 180
             Define how many segments are needed for drawing the ellipse.
-            The drawing will be smoother if you have many segments.
+            You can use this property to create polygons with 3 or more sides.
+            The default value is calculated from the range between angle.
```

Comment:
Since it defaults to `180`, why not using the default value (`180`) also if `segments < 3` ?

Reply:
We can, however the calculation of segments based on the angle has the advantage of being more efficient.

For example, if the angle range is 90 degrees then we will have 45 segments instead of 180, keeping a good visual appearance. The value of 180 was probably intended for a range of 360 degrees (default).

Reply:
Now the default value will be automatically calculated based on the angle range, as it is the most efficient approach 🙂